### DISCUSSION TITLE
Unable to return FT.Search results along with WithScores output

### DISCUSSION THREAD

**Discussion by hrshtsharma** [redis/rueidis#506] [Q&A]
Hey!

So I was recently trying to use the library to return the results for the query given below:
`FT.SEARCH myIndex "foo" SCORER BM25 WITHSCORES`

I am currently using the AsFTSearch Method to extract the result from the redisResult response. However, the results seems to the missing an entry for the scores that were returned. Inspecting the raw results, it seems that the results are part of the response but are being filtered out in the above process. I am currently extracting the results from the raw results directly but I believe that there should a better way to do this.

**Comment by rueian** [comment]
Hi @hrshtsharma,

I believe that there should be a better way to do this too! Could you share how do you get the scores?

**Reply by hrshtsharma** [reply]
@rueian Thanks for looking into it. Here is the snippet of the code being used. Currently, the scores are basically extracted and appended onto the FtSearch results.

```
_, searchResults, err := res.AsFtSearch()
respWithScores, _ := res.ToMap()
resultsWithScores := respWithScores["results"]
resultsArray, _ := resultsWithScores.ToArray()
for i, result := range resultsArray {
	r, _ := result.ToMap()
	rs := r["score"]
	rs2, _ := rs.ToFloat64()
	searchResults[i].Doc["score"] = strconv.FormatFloat(rs2, 'f', 4, 64)

```

**Reply by rueian** [reply]
Hi @hrshtsharma,

The next release will try to parse scores from a ftsearch response https://github.com/redis/rueidis/pull/509
